contracts/btc-finality/src/finality.rs

```diff
@@ -503,6 +509,11 @@ pub fn handle_finality_signature(
         let (msg, ev) = slash_finality_provider(&mut deps, &fp_btc_pk_hex, &evidence)?;
         res = res.add_message(msg);
         res = res.add_event(ev);
+    } else {
+        // Accumulate voting weight for this FP for reward distribution
+        ACCUMULATED_VOTING_WEIGHTS.update(deps.storage, &fp_btc_pk_hex, |existing| {
+            Ok::<u128, ContractError>(existing.unwrap_or(0) + (voting_power as u128))
+        })?;
     }
```

Comment:
This could be put between L495 and L497, not necessarily in this else statament

Reply:
Hmmm, then the fp that submits conflicting votes could receive rewards for the same height twice, no?

Reply:
The vote over the fork block will not because the function will return [here](https://github.com/babylonlabs-io/cosmos-bsn-contracts/blob/98711f8199d01e09ee8b94c71dcb2ca2c6f42a59/contracts/btc-finality/src/finality.rs#L449-L450). Is that correct?

Reply:
Maybe a cleaner way is the add return in the if statement so that we don't need else statement. Fixed in 
a41a675